File: Other/CheckPermutation.py
from nose.tools import assert_equal

def checkPerm(l1, l2):

	if len(l1) != len(l2):
		return False

	dic = {}

	for val in l1:
		if val in dic:
			dic[val] += 1
		else:
			dic[val] = 1

	for val in l2:
		if val in dic:
			dic[val] -= 1
			if dic[val] < 0:
				return False
		else:
			return False

	if sum(dic.values()) == 0:
		return True
	else:
		return False

class checkPermTest:
	def test(self, func):
		assert_equal(func('abc', 'bca'), True)
		assert_equal(func('abc', 'ca'), False)
		assert_equal(func('abc', 'cca'), False)
		print('TESTS PASSED')

t = checkPermTest()
t.test(checkPerm)

File: Other/CheckPermutation.js
function checkPerm(a1, a2){
	var hash = {};
	var sum = 0;

	for(i=0; i<a1.length; i++){
		if(hash[a1[i]] === undefined){
			hash[a1[i]] = 1;
		} else{
			hash[a1[i]]++;
		}
	}

	for(i=0; i<a2.length;i++){
		if(hash[a2[i]] === undefined){
			return false;
		} else {
			hash[a2[i]]--;
			if(hash[a2[i]] < 0){
				return false
			}
		}
	}

	for(key in hash){
		sum += hash[key]
	}

	if(sum === 0){
		return true;
	} else {
		return false;
	}
}

console.log(checkPerm('abc', 'bca'));
console.log(checkPerm('abc', 'bc'));
console.log(checkPerm('abc', 'cca'));